client/src/pages/course/admin/cross-check-table.tsx

```diff
@@ -86,8 +86,9 @@ export function Page(props: CoursePageProps) {
 
   const getCourseScore = useCallback(
     async (pagination: IPaginationInfo, filters: typeof fields, order: ScoreOrder) => {
+      const orderField = Array.isArray(order.field) ? order.field[0] : order.field;
```

Comment:
not good way create new crutch, if we don't fix reason why this situation happened
can you try delete `as any` in this row  ` onChange={handleChange as any}` number row is `173` and fixed problem with types.

Reply:
as far as I can tell `order.fields` comes from `dataIndex`, if it's an array so are fields, if it's not - fields not too.  We can use either this "crutch" or get rid of array in `dataIndex`, by mapping incoming data from `api.getCrossCheckPairs` or even better - changing that service on backend side, I've tried mapping incoming data + changing `dataIndex`, it works, but I don't think it's an improvement comparing to the first "crutch". 

When we use `dataIndex` as an array we instruct Table to look at nested props in data source. The `CrossCheckPairs` looks like this 
```ts
interface CrossCheckPairs {
  checker: {
    githubId: string;
    id: number;
  };
  task: {
    name: string;
    id: number;
  };
  student: {
    githubId: string;
    id: number;
  };
  url: string;
  comment: string;
  score: number;
  id: number;
}

```
therefore was used `dataIndex` like this: `["task", "name"]` but for ordering we only need `task`.
I mapped incoming data to 
```ts
interface CrossCheckPairs {
  checker: string;
  task: string
  student: string;
  url: string;
  comment: string;
  score: number;
  id: number;
}
```
and it works, but again it's probably not the solution we're looking for. 

So far this service is used only on this page, and I don't know if it was build only to be used here or it's going to be used somewhere else. 

So we can either check if `order.field` is an array, or change/add service to get data in more useful way, or maybe there's some other way to specify `order.field` which I'm not aware of, haven't found anything related on docs

Reply:
Why we do it in this case? 

> When we use dataIndex as an array we instruct Table to look at nested props in data source. 

If we use this service only on this page and for other tables we use ant sorting(without backend) in table, make sense use only `string` in `dataIndex` like `dataIndex: 'checker'` or `dataIndex: 'task'` in this case

Reply:
It was made this way 2 years ago. Anyway we can't just use `dataIndex` as a string without changing incoming data either on the client side or on the server side, because its main purpose is to access data to display it, sorting is like a side effect.

However I've just tried ant sorting `stringSorter('task.name')` and it seams like working, at least on the test db. 
So I'll leave `dataIndex` the way it is, adjust sorter and try to get rid of `any` type. 

Thanks for the tip.

Reply:
@ShekoMaxim Turns out using ant sorting isn't going to work here (same as on [Student Score table](https://github.com/rolling-scopes/rsschool-app/pull/1532)), because it works only with local data, it'll sort [within](https://github.com/rolling-scopes/rsschool-app/issues/1526) the current pagination - we'll get each pagination sorted separately, which is not what you'd expect - sorting the whole table. 

So we need to use sorting on the backend side which will return sorted and filtered data for current pagination. Therefore I fixed types, and added extra property for column type, where we can specify sorter field and use it on the request to the backend.

Any thoughts on this approach?